### ISSUE TITLE
Enhancing Device Management with Cockpit  and Custom Image Configuration

### ISSUE THREAD

**Issue by jastlw** [ublue-os/image-template#41]
Hi people.

The idea is for each organization to create its own image, allowing them to distribute their required software and customize the following settings:

1. YubiKey Configuration for LUKS Encryption:
        Is it possible to preconfigure YubiKeys for unlocking LUKS encryption for all employees or just for admins?

2. Pre-configured Admin Account:
        The image should also include a pre-configured admin account with a public key for SSH or YubiKey login.

3. Domain Login Configuration:
        Can we configure the image to allow domain login (LDAP or Microsoft Entra) with the necessary trust relationship settings?

4. Automatic SSH Tunnel Configuration:
        Is it feasible to set up an automatically opening SSH tunnel (upon network change) to an organizational server, enabling admins to manage devices via Cockpit through this tunnel? The public key of the organizational server would be embedded in the image.

5. Additional: VPN Pre-Configuration:
        Could we pre-configure a VPN connection that users can start with their username and password?

Additionally, I would like to note that Cockpit already supports remote management (see [Issue](https://github.com/ublue-os/bluefin/issues/271)).

Does this functionality already exist, and do you think these configurations are possible and practical? What do you think? I am a newbie and your insights would be greatly appreciated.

Thank you!

**Comment by castrojo** [comment]
Yeah you can do just about anything you want, I recommend taking whatever procedures/scripts you have to do that and adapt them into a containerfile.

**Comment by castrojo** [comment]
There's a fresh new bootc discussion forum where people who will be building custom images will be hanging out, you should repost there. I'll see you there!

https://github.com/containers/bootc/discussions

I'll add a more prominent link in the template.
